### PR TITLE
[14.0][FIX] stock_request: use stock request expected_date as deadline_date

### PR DESCRIPTION
**Pull request by petrus-v**
In this PR I've done 3 things, let me now It you prefer 3 distincts PRs:

* https://github.com/OCA/stock-logistics-warehouse/commit/5f40bc0f997c18b73e81f1049cfb47291a2f1a58: improves unit test to make sure `assetRaises` realy test what it expected to test
* https://github.com/OCA/stock-logistics-warehouse/commit/dddabb3003781f83a80cf45791428df5f45fc3e9: do not force default expected date on stock.request if provide on creation without stock.request.order
* use the expected date as deadline_date on procurement creation to avoid merging `stock.move` from different stock.request with different expected date. I wonder the impact on running prod env and wonder if I should do something less impacting ?

### CODE REVIEW

File: stock_request/models/stock_request.py [hunk 1]
```diff
@@ -283,6 +280,7 @@ def _prepare_procurement_values(self, group_id=False):
         """
         return {
             "date_planned": self.expected_date,
+            "date_deadline": self.expected_date,
```
Comment (petrus-v) [inline stock_request/models/stock_request.py:333]: I would prefer send date instead date time but it really depends on business here :thinking:

Comment (petrus-v) [reply stock_request/models/stock_request.py:333]: In fact that this is an other matter that should be address from an other module, this proposed fix is similar from what's  done in sale_stock module while selling goods, date_deadline  is set from date_order which is a datetime

Comment (petrus-v) [reply stock_request/models/stock_request.py:333]: done by https://github.com/OCA/stock-logistics-workflow/pull/1226

**Comment by petrus-v** [comment]
@hparfr, @Kev-Roche review welcome here !

File: stock_request/models/stock_request.py [hunk 2]
```diff
@@ -48,6 +44,7 @@ def _get_expected_date():
     expected_date = fields.Datetime(
         "Expected Date",
         index=True,
+        default=lambda self: fields.Datetime.now(),
```
Comment (rousseldenis) [inline stock_request/models/stock_request.py]: @petrus-v I would have put this in a function in order to allow override.

File: stock_request/tests/test_stock_request.py [hunk 3]
```diff
@@ -451,7 +492,7 @@ def test_stock_request_order_validations_06(self):
         expected_date = fields.Datetime.now()
         child_expected_date = "2015-01-01"
         vals = {
-            "company_id": self.company_2.id,
+            "company_id": self.main_company.id,
```
Comment (rousseldenis) [inline stock_request/tests/test_stock_request.py:497]: Why this change ?

File: stock_request/models/stock_request.py (hunk 2 above)
Comment (petrus-v) [reply stock_request/models/stock_request.py]: I agree there is no reason for this change !...

File: stock_request/tests/test_stock_request.py (hunk 3 above)
Comment (petrus-v) [reply stock_request/tests/test_stock_request.py:497]: I probably think this `test_stock_request_order_validations_06` was the same test as [test_stock_request_order_validations_05](https://github.com/OCA/stock-logistics-warehouse/pull/1637/files#diff-aea988ceb44324ce797bd2eaf94bc771996f7db0ef6308404d2bf431570ec55cR430), but maybe not... I'll probably restore this test and create a new one changing the doc-string seems to be outdated.

Thanks

File: stock_request/models/stock_request.py (hunk 2 above)
Comment (petrus-v) [reply stock_request/models/stock_request.py]: hum.. I mean I'll use the existing method !

File: stock_request/tests/test_stock_request.py (hunk 3 above)
Comment (petrus-v) [reply stock_request/tests/test_stock_request.py:497]: I've rename the test and change the docstring, I confirm that it was testing the same assertion as `test_stock_request_order_validations_05` !

File: stock_request/models/stock_request.py (hunk 2 above)
Comment (petrus-v) [reply stock_request/models/stock_request.py]: done

**Review by damdam-s — APPROVED** [review]
